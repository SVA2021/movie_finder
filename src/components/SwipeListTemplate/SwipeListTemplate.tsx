import NavigateBeforeIcon from '@mui/icons-material/NavigateBefore';
import NavigateNextIcon from '@mui/icons-material/NavigateNext';
import {Button, Grid, useTheme} from '@mui/material';
import {FC, useRef} from 'react';
import {Swiper, SwiperSlide} from 'swiper/react';
import 'swiper/swiper.min.css'; // core Swiper
import {EmptyDataBox} from '..';
import {SmallCard} from '../SmallCard/SmallCard';
import {SwipeListTemplateProps} from './SwiperListTemplate.types';

export const SwipeListTemplate: FC<SwipeListTemplateProps> = ({data}) => {
  const theme = useTheme();
  const swiperRef = useRef<any>();

  const swipePrev = () => swiperRef.current.slidePrev();
  const swipeNext = () => swiperRef.current.slideNext();

  return (
    <Grid container
      m={0}
      spacing={0}
      direction={'row'}
      alignItems={'center'}
      justifyContent={'space-between'}
      bgcolor={'primary.main'}
    >
      {data.length > 0
        ? <>
          <Grid item
            lg={1}
            justifyContent={'center'}
            sx={{display: {xs: 'none', lg: 'flex'}}}
          >
            <Button onClick={swipePrev}>
              <NavigateBeforeIcon fontSize='large' />
            </Button>
          </Grid>
          <Grid item xs={12} lg={10} >
            <Swiper
              onSwiper={(swiper) => swiperRef.current = swiper}
              slidesPerView={1}
              spaceBetween={8}
              breakpoints={
                {
                  [theme.breakpoints.values.md]: {
                    slidesPerView: 4,
                    spaceBetween: 16,
                  },
                  [theme.breakpoints.values.lg]: {
                    slidesPerView: 5,
                    spaceBetween: 24,
                  },
                }
              }
            >
              {
                data.map((item, index) =>
                  <SwiperSlide key={index} >
                    <SmallCard item={item} />
                  </SwiperSlide>
                )
              }
            </Swiper>
          </Grid>
          <Grid item
            lg={1}
            justifyContent={'center'}
            sx={{display: {xs: 'none', lg: 'flex'}}}
          >
            <Button onClick={swipeNext}>
              <NavigateNextIcon fontSize='large' />
            </Button>
          </Grid>
        </>
        :
        <Grid item xs={12}>
          <EmptyDataBox />
        </Grid>
      }
    </Grid>
  );
};