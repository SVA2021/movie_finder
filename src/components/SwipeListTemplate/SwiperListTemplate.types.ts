export interface SwipeListTemplateProps {
    data: any[]
}