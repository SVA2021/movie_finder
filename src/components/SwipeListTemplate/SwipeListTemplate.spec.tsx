import {render, screen} from "@testing-library/react";
import {BrowserRouter} from "react-router-dom";
import {SwipeListTemplate} from "./SwipeListTemplate";

const FakeCard = {
  "id": 325381,
  "nameRu": "fakeNameRu",
  "nameEn": "fakeNameEng",
  "year": "2000",
  "filmLength": "10:10",
  "rating": "7.0",
  "posterUrlPreview": "fakePosterUrlPreview",
}

describe('render SwipeListTemplate', () => {

  const setUp = () => render(
    <BrowserRouter>
      <SwipeListTemplate data={[FakeCard]} />
    </BrowserRouter>
  );
  const emptySetUp = () => render(
    <BrowserRouter>
      <SwipeListTemplate data={[]} />
    </BrowserRouter>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('swipe-list-template')).toMatchSnapshot();
  });

  it('check text in card', () => {
    setUp();
    expect(screen.getByText(/fakeNameRu/i)).toBeInTheDocument();
  });

  it('check empty data placeholder', () => {
    emptySetUp();
    expect(screen.getByText(/нет данных/i)).toBeInTheDocument();
  });

})