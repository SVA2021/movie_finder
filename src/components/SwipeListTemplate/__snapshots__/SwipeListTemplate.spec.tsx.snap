// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SwipeListTemplate snapshot test 1`] = `
<div
  class="MuiGrid-root MuiGrid-container css-1nsfak8-MuiGrid-root"
  data-testid="swipe-list-template"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-1 css-p038cf-MuiGrid-root"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="swipe-list-prevbtn"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="NavigateBeforeIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-10 css-1hlpjpi-MuiGrid-root"
  >
    <div
      class="swiper swiper-initialized swiper-horizontal swiper-pointer-events"
    >
      <div
        class="swiper-wrapper"
        style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-4ob670-MuiPaper-root-MuiCard-root"
            data-testid="smallCard"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root css-tqw9zv-MuiButtonBase-root-MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <img
                alt="fakeNameRu"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                src="fakePosterUrlPreview"
                width="100%"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1ydyjjr-MuiTypography-root"
                >
                  fakeNameRu
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-12e029l-MuiTypography-root"
                >
                  fakeNameEng
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                >
                  2000
                  ,  
                  10ч10м
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <p
              class="MuiTypography-root MuiTypography-h3 css-1uovp3e-MuiTypography-root"
              data-testid="rating-badge"
            >
              7.0
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-1 css-p038cf-MuiGrid-root"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="swipe-list-nextbtn"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="NavigateNextIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
